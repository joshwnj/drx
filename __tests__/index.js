/* global describe, it, expect */

const x = require('../src')
const renderer = require('react-test-renderer')
const { createElement } = require('react')

function render (type, props) {
  return renderer.create(
    createElement(type, props)
  ).toJSON()
}

describe('Classes', () => {
  it('with no classes', () => {
    const Component = x.div('some-class')
    expect(render(Component)).toMatchSnapshot()
  })

  it('with a single class', () => {
    const Component = x.div('some-class')
    expect(render(Component)).toMatchSnapshot()
  })

  it('with multiple classes', () => {
    const Component = x.div('some-class', 'another-class')
    expect(render(Component)).toMatchSnapshot()
  })
})

describe('Div with a conditional class', () => {
  const Component = x.div(
    'thing',
    p => p.big ? 'thing--big' : [ 'thing--small', 'small' ]
  )

  it('gets a single extra class with the prop flag', () => {
    expect(render(Component, { big: true })).toMatchSnapshot()
  })

  it('gets two extra classes without the prop flag', () => {
    expect(render(Component)).toMatchSnapshot()
  })
})

describe('Basic conditional rendering', () => {
  const Component = x.div()
    .renderIf('active')
    .props('children')

  it('with the prop flag', () => {
    expect(
      render(Component, { active: true, children: 'hi' })
    ).toMatchSnapshot()
  })

  it('without the prop flag', () => {
    expect(
      render(Component, { children: 'hi' })
    ).toMatchSnapshot()
  })
})

describe('Conditional rendering by function', () => {
  const Component = x.div()
    .renderIf(p => p.a + p.b > 10)
    .props('children')

  it('with sum of props greater than 10', () => {
    expect(
      render(Component, { a: 9, b: 2, children: 'hi' })
    ).toMatchSnapshot()
  })

  it('with sum of props less than 10', () => {
    expect(
      render(Component, { a: 1, b: 8, children: 'hi' })
    ).toMatchSnapshot()
  })
})

describe('Passing props', () => {
  const Image = x.img()
    .props('src', 'alt')

  const Text = x.span()
    .props('children')

  const Component = x.div()
    .content(
      Text,
      Image
    )

  it('passes props to child components', () => {
    expect(
      render(Component, { src: 'awyis.gif', alt: 'aw yis', children: 'a gif' })
    ).toMatchSnapshot()
  })
})
