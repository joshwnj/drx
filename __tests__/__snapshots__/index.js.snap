// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic conditional rendering with the prop flag 1`] = `
<div>
  hi
</div>
`;

exports[`Basic conditional rendering without the prop flag 1`] = `null`;

exports[`Classes with a single class 1`] = `
<div
  className="some-class"
/>
`;

exports[`Classes with multiple classes 1`] = `
<div
  className="some-class another-class"
/>
`;

exports[`Classes with no classes 1`] = `
<div
  className="some-class"
/>
`;

exports[`Conditional rendering by function with sum of props greater than 10 1`] = `
<div>
  hi
</div>
`;

exports[`Conditional rendering by function with sum of props less than 10 1`] = `null`;

exports[`Div with a conditional class gets a single extra class with the prop flag 1`] = `
<div
  className="thing thing--big"
/>
`;

exports[`Div with a conditional class gets two extra classes without the prop flag 1`] = `
<div
  className="thing thing--small small"
/>
`;

exports[`Passing props passes props to child components 1`] = `
<div>
  <span>
    a gif
  </span>
  <img
    alt="aw yis"
    src="awyis.gif"
  />
</div>
`;
